### ISSUE TITLE
autocorrection of [Style/IfUnlessModifier] is wrong when assigning value

### ISSUE THREAD

**Issue by akostadinov** [rubocop/rubocop#12057]
I wanted to assign a variable based on a condition.

```ruby
if condition
  @match = compute(input)
else
  @match = nil
end
```

Then rubocop auto-corrected that to
```ruby
@match = if condition
  compute(input)
else
  nil
end
```

But then complained that the `else` condition was useless. Which is true so **I** changed it to
```ruby
@match = if condition
  compute(input)
end
```

And then Rubocop autocorrects it to:
```ruby
@match = compute(input) if condition
```

Bow here is the specific bug. That the semantics is very different between
```ruby
@match = compute(input) if condition
@match = if condition; compute(input); end
```
Because in the latter case the value of `@match` will be reset to `nil` when condition is `falsy`. While `@match` keeps its previous value in the former case.

So the auto-correction should not be performed when assigning to a variable.

--------

## Expected behavior
No offense reported 

## Actual behavior
Offense reported and wrong auto-correction applied

## Steps to reproduce the problem
See description.

## RuboCop version

```
1.54.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.18.0
```

**Comment by fatkodima** [comment]
Tested on master. 
```ruby
@match = if condition
  compute(input)
end
```

is autocorrected to
```ruby
@match = (compute(input) if condition)
```
which, thanks to parens, is equal to `condition ? compute(input) : nil` and works correctly.

Am I missing something?

**Comment by akostadinov** [comment]
Ok, you're right, I didn't notice the parenthesis. On the other hand it is hugely annoying also that with these transformations, it creates very long lines which don't improve readability.

Thanks for the answer!
